models/spring-ai-postgresml/src/main/java/org/springframework/ai/postgresml/PostgresMlEmbeddingClient.java

```diff
@@ -110,73 +150,98 @@ public PostgresMlEmbeddingClient(JdbcTemplate jdbcTemplate, String transformer,
-			preparedStatement.setString(1, transformer);
-			preparedStatement.setString(2, kwargs);
-			preparedStatement.setArray(3, connection.createArrayOf("TEXT", texts.toArray(Object[]::new)));
-			return preparedStatement;
-		}, rs -> {
-			List<List<Double>> result = new ArrayList<>();
-			while (rs.next()) {
-				result.add(vectorType.rowMapper.mapRow(rs, -1));
-			}
-			return result;
-		});
-	}
+	public EmbeddingResponse call(EmbeddingRequest request) {
 
-	@Override
-	public EmbeddingResponse embedForResponse(List<String> texts) {
-		return this.call(new EmbeddingRequest(texts, EmbeddingOptions.EMPTY));
-	}
+		final PostgresMlEmbeddingOptions optionsToUse = this.mergeOptions(request.getOptions());
 
-	@Override
-	public EmbeddingResponse call(EmbeddingRequest request) {
 		List<Embedding> data = new ArrayList<>();
-		List<List<Double>> embed = this.embed(request.getInstructions());
-		for (int i = 0; i < embed.size(); i++) {
-			data.add(new Embedding(embed.get(i), i));
+		List<List<Double>> embed = List.of();
+
+		List<String> texts = request.getInstructions();
+		if (!CollectionUtils.isEmpty(texts)) {
+			embed = this.jdbcTemplate.query(connection -> {
+				PreparedStatement preparedStatement = connection.prepareStatement("SELECT pgml.embed(?, text, ?::JSONB)"
+						+ optionsToUse.getVectorType().cast + " AS embedding FROM (SELECT unnest(?) AS text) AS texts");
+				preparedStatement.setString(1, optionsToUse.getTransformer());
+				preparedStatement.setString(2, ModelOptionsUtils.toJsonString(optionsToUse.getKwargs()));
+				preparedStatement.setArray(3, connection.createArrayOf("TEXT", texts.toArray(Object[]::new)));
+				return preparedStatement;
+			}, rs -> {
+				List<List<Double>> result = new ArrayList<>();
+				while (rs.next()) {
+					result.add(optionsToUse.getVectorType().rowMapper.mapRow(rs, -1));
+				}
+				return result;
+			});
+		}
+
+		if (!CollectionUtils.isEmpty(embed)) {
+			for (int i = 0; i < embed.size(); i++) {
+				data.add(new Embedding(embed.get(i), i));
+			}
 		}
+
 		var metadata = new EmbeddingResponseMetadata(
-				Map.of("transformer", this.transformer, "vector-type", this.vectorType.name(), "kwargs", this.kwargs));
+				Map.of("transformer", optionsToUse.getTransformer(), "vector-type", optionsToUse.getVectorType().name(),
+						"kwargs", ModelOptionsUtils.toJsonString(optionsToUse.getKwargs())));
 
 		return new EmbeddingResponse(data, metadata);
 	}
 
+	/**
+	 * Merge the default and request options.
+	 * @param requestOptions request options to merge.
+	 * @return the merged options.
+	 */
+	PostgresMlEmbeddingOptions mergeOptions(EmbeddingOptions requestOptions) {
+
+		PostgresMlEmbeddingOptions options = PostgresMlEmbeddingOptions.builder()
+			.withTransformer(DEFAULT_TRANSFORMER_MODEL)
+			.build();
+
+		if (this.defaultOptions != null) {
+			options = ModelOptionsUtils.merge(this.defaultOptions, options, PostgresMlEmbeddingOptions.class);
+		}
+
+		if (requestOptions != null) {
```

Comment:
`ModelOptionsUtils.merge` can be skipped when `requestOptions` is `EmbeddingOptions.EMPTY`.

```java
if (requestOptions != null && !EmbeddingOptions.EMPTY.equals(requestOptions))
```

Reply:
same as above.